src/EmptyState/EmptyState.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+import classNames from 'classnames';
+import propTypes from 'prop-types';
+
+import { Heading } from 'src/Heading';
+import { Text } from 'src/Text';
+
+import './EmptyState.scss';
+
+export const EmptyStateMarginTopSizes = {
+  SMALL: 'sm',
+  MEDIUM: 'md',
+  LARGE: 'lg',
+  NONE: 'none',
+};
+
+const EmptyState = ({
+  className,
+  fullWidth,
+  marginTop,
+  primaryAction,
+  subtitle,
+  title,
+}) => (
+  <div className={classNames(
+    className,
+    'EmptyState',
+    marginTop && `EmptyState--margin-top--${marginTop}`,
+  )}
+  >
+    <div className={classNames(
+      'EmptyState__content',
+      fullWidth && 'EmptyState--full-width',
+    )}
+    >
+      {title && (
+        <>
+          { typeof title === 'string' ? (
+            <Heading className="EmptyState__title" level={4} size="lg" textAlign="center">{title}</Heading>
+          ) : (
+            <>{title}</>
+          )}
```

Comment:
not really sure if this overdoing it trying to build in some flexibility just in case `title` isn't just a string and you might need to put some other element (e.g. a link with some text or tooltip). I could err on the side of enforcement of just a string here though if that makes more sense.

Reply:
```suggestion
          <Heading className="EmptyState__title" level={4} size="lg" textAlign="center">{title}</Heading>
```

My instinct is to not do the string check, always wrap it in the heading, but have the propType be `elementType` so it can take a string, a function/component, or an element-like object such as a fragment.

Then if you want formatting you could pass a component as the title or do something like below to get additional formatting, without giving up the default styles. 

`<EmptyState title={<>This title has <em>italics</em> and a <a href="www.google.com">link</a></>}`

I think enforcing the font size, text align etc is good for consistency. Otherwise we could end up with everyone using this component but with the heading size/styling as inconsistent as it is now.

I could see a `toolTipText` prop being useful as well.

Reply:
Thanks I'll err on the side of enforcement. That makes sense.

If found this on the difference between `node` and `elementType`:

PropTypes.node is used to specify that a prop can accept any valid React node as its value. A React node represents any valid JSX expression, including strings, numbers, React elements, arrays, fragments, and portals. This means that the prop can receive components, HTML elements, text content, or any other valid JSX representation.

PropTypes.elementType is used to specify that a prop should receive a React component (class or functional) as its value. It ensures that the prop accepts only a valid React component type and not other types like strings, numbers, or elements.

I think I've found myself using `node` more often just for the flexibility unless I know for sure I need a component in `elementType`.